cpp/core/compute/ExecutionCtx.h

```diff
@@ -47,11 +48,18 @@ class ExecutionCtx : public std::enable_shared_from_this<ExecutionCtx> {
   ExecutionCtx(const std::unordered_map<std::string, std::string>& confMap) : confMap_(confMap) {}
   virtual ~ExecutionCtx() = default;
 
-  virtual std::shared_ptr<ResultIterator> getResultIterator(
+  virtual ResourceHandle createResultIterator(
       MemoryManager* memoryManager,
       const std::string& spillDir,
       const std::vector<std::shared_ptr<ResultIterator>>& inputs,
       const std::unordered_map<std::string, std::string>& sessionConf) = 0;
+  virtual ResourceHandle addResultIterator(std::shared_ptr<ResultIterator>) = 0;
+  virtual std::shared_ptr<ResultIterator> getResultIterator(ResourceHandle) = 0;
```

Comment:
Should we make the `add...` member funcs private?

Reply:
`addResultIterator` is needed for legacy codes, we may refactor later.